pkg/component/watch_test.go

```diff
@@ -68,31 +73,86 @@ func setUpF8AnalyticsComponentSrc(componentName string, requiredFilePaths []test
 	return srcPath, retVal, nil
 }
 
-// ExpectedChangedFiles is required so that the mockPushLocal below can validate obrtained change set against the test expected changes
+// ExpectedChangedFiles is required so that the mockPushLocal below can validate obtained changes set against the test expected changes
 var ExpectedChangedFiles []string
 
+// DeleteFiles is required to validate deleted changes set against the test expected changes
+var DeleteFiles []string
+
 // CompDirStructure is required to hold the directory structure of mock component created by the test which can be accessed by mockPushLocal
-var CompDirStructure map[string]testingutil.FileProperties
+var (
+	muLock           sync.Mutex
+	CompDirStructure map[string]testingutil.FileProperties
+)
 
 // ExtChan is used to return from otherwise non-terminating(without SIGINT) end of ever running watch function
 var ExtChan = make(chan bool)
 var StartChan = make(chan bool)
 
+type mockPushParameters struct {
+	componentName   string
+	applicationName string
+	path            string
+	isForcePush     bool
+	globExps        []string
+	show            bool
+}
+
+var mockPush mockPushParameters
+
 // Mock PushLocal to collect changed files and compare against expected changed files
 func mockPushLocal(client *occlient.Client, componentName string, applicationName string, path string, out io.Writer, files []string, delFiles []string, isPushForce bool, globExps []string, show bool) error {
-	for _, gotChangedFile := range files {
+	muLock.Lock()
+	defer muLock.Unlock()
+	if componentName != mockPush.componentName || applicationName != mockPush.applicationName || isPushForce != mockPush.isForcePush || show != mockPush.show {
+		fmt.Printf("some of the push parameters are different, wanted: %v, got: %v", mockPush, []string{
+			componentName, applicationName, "isPushForce:" + strconv.FormatBool(isPushForce), "show:" + strconv.FormatBool(show),
+		})
+		os.Exit(1)
```

Comment:
Shouldn't this return error, rather than calling Exit?

Reply:
we are ignoring the error later on in the code, so the test passes if I don't exit